pkg/ebpf/dns_cache.go

```diff
@@ -102,12 +113,18 @@ func (c *reverseDNSCache) Get(conns []ConnectionStats, now time.Time) map[util.A
 			return
 		}
 
-		if names := c.getNamesForIP(addr, expiration); names != nil {
-			resolved[addr] = names
+		if _, ok := oversized[addr]; ok {
 			return
 		}
 
-		unresolved[addr] = struct{}{}
+		names := c.getNamesForIP(addr, expiration)
+		if len(names) == 0 {
+			unresolved[addr] = struct{}{}
+		} else if len(names) >= c.maxDomainsPerIP {
```

Comment:
Can `len(names) > c.maxDomainsPerIP` happen if we are maintaining the size during insertion?

Reply:
Good point! Done

Reply:
The `>` is still there. 
I am wondering if we should do this check here at all. My understanding is that we are letting the number of domains grow *upto* `maxDomainsPerIP` (line 217 below) but then discarding here any domain list whose length is `maxDomainsPerIP`. We are discarding _all_ domains in that case. In essence, we are imposing the constraint that number of domains must be **less** than `maxDomainsPerIP` for them to be returned to the tracer. 

Instead, can we just return whatever domains we have since the `merge` function ensures `maxDomainsPerIP` constraint is being maintained and increase some counter there everytime we reject a value because of that constraint? Am I missing something here?